### PR TITLE
refactor/chore: update cppcheck to 2.8 with needed refactoring

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
Seems like 2.8 doesn't have many false positives in our codebase.. Upgraded, added two exclusions for some false positives, and fixed other issues found by cppcheck

### CODE REVIEW

File: src/llmq/dkgsession.cpp [hunk 1]
```diff
@@ -372,13 +372,13 @@ void CDKGSession::VerifyPendingContributions()
...
         vvecs.emplace_back(receivedVvecs[idx]);
-        skContributions.emplace_back(receivedSkContributions[idx]);
+        skContributions2.emplace_back(receivedSkContributions[idx]);
         // Write here to definitely store one contribution for each member no matter if
         // our share is valid or not, could be that others are still correct
         dkgManager.WriteEncryptedContributions(params.type, m_quorum_base_block_index, m->dmn->proTxHash, *vecEncryptedContributions[idx]);
     }
 
-    auto result = blsWorker.VerifyContributionShares(myId, vvecs, skContributions);
+    auto result = blsWorker.VerifyContributionShares(myId, vvecs, skContributions2);
```
Comment (UdjinM6) [inline src/llmq/dkgsession.cpp:381]: Missing s/`skContributions[i]`/`skContributions2[i]`/ at line 398 below. Also, smth2 is not very informative ("why 2?"). Let's maybe rename the class member instead e.g. `m_sk_contributions`?
>- Variable and namespace names are all lowercase, and may use _ to separate words (snake_case).
>    -  Class member variables have a m_ prefix.

https://github.com/dashpay/dash/blob/master/doc/developer-notes.md#coding-style-c

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/llmq/signing_shares.h [hunk 2]
```diff
@@ -35,7 +35,7 @@ constexpr uint32_t UNINITIALIZED_SESSION_ID{std::numeric_limits<uint32_t>::max()
 class CSigShare : virtual public CSigBase
 {
 protected:
-    uint16_t quorumMember;
+    uint16_t quorumMember{0};
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply src/llmq/signing_shares.h:38]: why is that better than 0?

File: src/llmq/dkgsession.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/llmq/dkgsession.cpp:381]: That would be a good solution :) I'll do that later (feel free to push a commit for it if you like)

File: src/llmq/signing_shares.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (UdjinM6) [reply src/llmq/signing_shares.h:38]: `0` is a valid index. By initialising it with smth invalid we would make sure it was actually assigned somewhere.

File: src/ctpl_stl.h [hunk 3]
```diff
@@ -129,7 +129,7 @@ namespace ctpl {
         std::function<void(int)> pop() {
             std::function<void(int id)> * _f = nullptr;
             this->q.pop(_f);
-            std::unique_ptr<std::function<void(int id)>> func(_f); // at return, delete the function even if an exception occurred
+            [[maybe_unused]] std::unique_ptr<std::function<void(int id)>> func(_f); // at return, delete the function even if an exception occurred
```
[the remark this answers is not in the document]
Comment (UdjinM6) [reply src/ctpl_stl.h:132]: ok :)

File: src/llmq/dkgsession.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/llmq/dkgsession.cpp:381]: Done :)

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Pull request merged**